artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java

```diff
@@ -289,6 +289,12 @@ public SimpleString getGroupID() {
       return this.getSimpleStringProperty(Message.HDR_GROUP_ID);
    }
 
+   @Override
+   public int getGroupSequence() {
+      final Integer integer = this.getIntProperty(Message.HDR_GROUP_SEQUENCE);
```

Comment:
This just blew up for me. It is not Mike's bug per se, it just possibly sends execution to this little function which cannot handle `null`s

https://github.com/apache/activemq-artemis/blob/10ecb358cb587c8417deeb799723d8274b0a44a5/artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/TypedProperties.java#L211-L214

In my JVM,

```
    public static Integer valueOf(String var0) throws NumberFormatException {
        return parseInt(var0, 10);
    }
```

and

```
    public static int parseInt(String var0, int var1) throws NumberFormatException {
        if (var0 == null) {
            throw new NumberFormatException("null");
```

Here is a failing test (from cli-java project) on travis ci, https://travis-ci.org/rh-messaging/cli-java/jobs/451880325 at around

> 13:28:41,911 ERROR Error while sending a message!
> java.lang.NumberFormatException: null

It was not supposed to fetch a 2.7.0 snapshot of artemis-cli-java, but somehow it did... mystery for the next time, for me or @michalxo.

Reply:
The regression introduced new, is fixed by https://github.com/apache/activemq-artemis/pull/2418

So the new regressions will go if/when that pr is merged